py/client/pydeephaven/session.py

```diff
@@ -127,12 +138,37 @@ def get_ticket(self):
 
             return self._last_ticket
 
+    def subscribe_fields(self):
+        """
+        Allows tables created/removed on the server to be automatically seen on the client
+        """
+
+        with self._r_lock:
+            if self._field_update_thread is None:
+                self._field_update_thread = threading.Thread(target=self._update_fields)
+                self._field_update_thread.daemon = True
+                self._field_update_thread.start()
+
+    def _update_fields(self):
+        """
+        Constant loop that checks for any server-side table changes and adds them to the local list
+        """
+
+        while True:
+            fields_change = next(self._list_fields)
+            with self._r_lock:
+                self._parse_fields_change(fields_change)
+
     def _connect(self):
         with self._r_lock:
             self.grpc_channel, self.session_token, self._timeout = self.session_service.connect()
             self.is_connected = True
+
             if self._never_timeout:
                 self._keep_alive()
+            
+            self._list_fields = self.app_service.list_fields()
+            self._parse_fields_change(next(self._list_fields))
```

Comment:
Couple things:

1. I don't think we should be subscribing on connect.
2. As is, it's not correct - we've starting a list_fields subscription, but we haven't started the threading handler for it. I don't know exactly how python gRPC will handle this (either a memory leak in list_fields, or eventually an error b/c it's not being consumed).

I think we need this logic self-contained within the subscribe_fields method.